src/ROCrate/Generic/PropertyValue.fs

```diff
@@ -0,0 +1,171 @@
+        | Some (:? string as n) -> n
+        | Some _ -> failwith $"Property of `name` of object with @id `{pv.Id}` was not a string"
+        | _ -> failwith $"Could not access property `name` of object with @id `{pv.Id}`"
+
+    static member setNameAsString(pv : LDNode, name : string, ?context : LDContext) =
+        pv.SetProperty(PropertyValue.name, name, ?context = context)
+
+    static member tryGetValueAsString(pv : LDNode, ?context : LDContext) =
+        match pv.TryGetPropertyAsSingleton(PropertyValue.value, ?context = context) with
+        | Some (:? string as v) -> Some v
+        | _ -> None
+
+    static member getValueAsString(pv : LDNode, ?context : LDContext) =
+        match pv.TryGetPropertyAsSingleton(PropertyValue.value, ?context = context) with
+        | Some (:? string as v) -> v
+        | Some _ -> failwith $"Property of `value` of object with @id `{pv.Id}` was not a string"
+        | _ -> failwith $"Could not access property `value` of object with @id `{pv.Id}`"
+
+    static member setValueAsString(pv : LDNode, value : string, ?context : LDContext) =
+        pv.SetProperty(PropertyValue.value, value, ?context = context)
+
+    static member tryGetPropertyIDAsString(pv : LDNode, ?context : LDContext) =
+        match pv.TryGetPropertyAsSingleton(PropertyValue.propertyID, ?context = context) with
+        | Some (:? string as pid) -> Some pid
+        | _ -> None
+
+    static member setPropertyIDAsString(pv : LDNode, propertyID : string, ?context : LDContext) =
+        pv.SetProperty(PropertyValue.propertyID, propertyID, ?context = context)
+
+    static member tryGetUnitCodeAsString(pv : LDNode, ?context : LDContext) =
+        match pv.TryGetPropertyAsSingleton(PropertyValue.unitCode, ?context = context) with
+        | Some (:? string as uc) -> Some uc
+        | _ -> None
+
+    static member setUnitCodeAsString(pv : LDNode, unitCode : string, ?context : LDContext) =
+        pv.SetProperty(PropertyValue.unitCode, unitCode, ?context = context)
+
+    static member tryGetUnitTextAsString(pv : LDNode, ?context : LDContext) =
+        match pv.TryGetPropertyAsSingleton(PropertyValue.unitText, ?context = context) with
+        | Some (:? string as ut) -> Some ut
+        | _ -> None
+
+    static member setUnitTextAsString(pv : LDNode, unitText : string, ?context : LDContext) =
+        pv.SetProperty(PropertyValue.unitText, unitText, ?context = context)
+
+    static member tryGetValueReference(pv : LDNode, ?context : LDContext) =
+        match pv.TryGetPropertyAsSingleton(PropertyValue.valueReference, ?context = context) with
+        | Some (:? string as vr) -> Some vr
+        | _ -> None
+
+    static member setValueReference(pv : LDNode, valueReference : string, ?context : LDContext) =
+        pv.SetProperty(PropertyValue.valueReference, valueReference, ?context = context)
+
+    static member validate(pv : LDNode, ?context : LDContext) =
```

Comment:
Is this missing validate functions for PropertyValue - DOI and PropertyValue - PubMedID ?

Reply:
There's currently not a validate function for doi and pubmedID.

The convertes can be found here:
https://github.com/nfdi4plants/ARCtrl/blob/b755341a2b50cc7e9243c95e55ed423e6e63cfa8/src/ARCtrl/Conversion.fs#L1015